app/2.0/docs/devguide/custom-elements.md

```diff
@@ -159,25 +129,28 @@ For a complete list of limitations, see [Requirements for custom element constru
 
 Whenever possible, defer work until the `connectedCallback` or later instead of performing it in the constructor.
 
-### One-time initialization
+### Polymer element initialization
 
 The custom elements specification doesn't provide a one-time initialization callback. Polymer
 provides a `ready` callback, invoked the first time the element is added to the DOM.
 
 ```js
 ready() {
   super.ready();
-  // When possible, use afterNextRender to defer non-critical
-  // work until after first paint.
-  Polymer.RenderStatus.afterNextRender(this, function() {
-    ...
-  });
-}
-```
+  // do something that requires access to the shadow tree
+  ... 
 
+}
+``` 
 
 The `Polymer.Element` class initializes your element's template and data system during the `ready`
-callback, so if you override ready, you must call `super.ready()` at some point.
+callback, so if you override `ready`, you must call `super.ready()` at some point.
+
+Polymer does several things at `ready` time:
+
+-   Sets default property values.
+-   Allows observers and computed properties to run (as soon as their dependencies are defined).
```

Comment:
In 2.0+ it's technically "as soon as _any_ of their dependencies are defined"

Reply:
OK, I think I was a little confused. I thought it was waiting until default values were set, but after experimenting a little more, it appears that defaults are now set in `constructor`, so we're just guaranteeing that the observers won't run until after that's happened, right?

Do we set properties (i.e., from initial bindings or attributes) before we let the observers fire?